zds/utils/templatetags/interventions.py

```diff
@@ -78,10 +79,21 @@ def humane_delta(value):
 
 @register.filter('followed_topics')
 def followed_topics(user):
-    topics_followed = TopicAnswerSubscription.objects.filter(user=user,
-                                                             content_type__model='topic',
-                                                             is_active=True)\
-        .order_by('-last_notification__pubdate')[:10]
+    """
+    gets the ten last followed topics.
+    @TODO : make the "10" parameter a configuration one
+    :param user:
+    :return:
+    """
+    topics_followed = TopicAnswerSubscription.objects\
+                        .fetch_related("content_object",
+                                       "content_object__last_message")\
+                        .extra(sort_date=F("last_notification__pubdate"))\
+                        .extra(displayed_date=F("content_object__last_message__pubdate"))\
```

Comment:
c'est un nom de django : F pour "field" et Q pour "sub query". Désolé,
c'est du django pur, j'aime pas non plus perso.

2016-05-30 14:37 GMT+02:00 Gérard Paligot <notifications@github.com>:

> In zds/utils/templatetags/interventions.py
> <https://github.com/zestedesavoir/zds-site/pull/3643#discussion_r65065992>
> :
>
> > @@ -78,10 +79,21 @@ def humane_delta(value):
> >
> >  @register.filter('followed_topics')
> >  def followed_topics(user):
> > -    topics_followed = TopicAnswerSubscription.objects.filter(user=user,
> > -                                                             content_type__model='topic',
> > -                                                             is_active=True)\
> > -        .order_by('-last_notification__pubdate')[:10]
> > +    """
> > +    gets the ten last followed topics.
> > +    @TODO : make the "10" parameter a configuration one
> > +    :param user:
> > +    :return:
> > +    """
> > +    topics_followed = TopicAnswerSubscription.objects\
> > +                        .fetch_related("content_object",
> > +                                       "content_object__last_message")\
> > +                        .extra(sort_date=F("last_notification__pubdate"))\
> > +                        .extra(displayed_date=F("content_object__last_message__pubdate"))\
>
> Bon à savoir et F sert à quoi ? (Ce nom ne veut juste rien dire ...-
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zestedesavoir/zds-site/pull/3643/files/27c1f079d3238b57b64613e3be47dea44bf8aee4#r65065992>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABc_xask52sxc-gihPR5HZpZQCOtHGlTks5qGtn1gaJpZM4Ipu2e>
> .
>